test/libcommon/log/sentry/testsentryhandler.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Infomaniak kDrive - Desktop
+ * Copyright (C) 2023-2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "config.h"
+#include "testsentryhandler.h"
+#include "libcommon/log/sentry/sentryhandler.h"
+#include "libcommonserver/log/log.h"
+
+#include <thread>
+
+namespace KDC {
+void MockTestSentryHandler::sendEventToSentry(const SentryLevel level, const std::string& title,
+                                              const std::string& message) const {
+    _sentryUploadedEventCount++;
+}
+
+
+void TestSentryHandler::testMultipleSendEventForTheSameEvent() {
+    MockTestSentryHandler mockSentryHandler;
+
+    CPPUNIT_ASSERT_EQUAL(0, mockSentryHandler.sentryUploadedEventCount());
+    mockSentryHandler.captureMessage(SentryLevel::Info, "Test", "Test message"); // Should be sent
+    CPPUNIT_ASSERT_EQUAL(1, mockSentryHandler.sentryUploadedEventCount());
+    mockSentryHandler.captureMessage(SentryLevel::Info, "Test", "Test message"); // Rate limit not reached, should be sent
+    CPPUNIT_ASSERT_EQUAL(2, mockSentryHandler.sentryUploadedEventCount());
+    mockSentryHandler.captureMessage(SentryLevel::Info, "Test",
+                                     "Test message"); // Rate limit reached, should be sent (first event on rate limit is sent).
+    CPPUNIT_ASSERT_EQUAL(3, mockSentryHandler.sentryUploadedEventCount());
+
+    for (int i = 3; i <= 4; i++) { // Done twice to check that the timer is well reset.
+        mockSentryHandler.captureMessage(SentryLevel::Info, "Test", "Test message"); // Rate limit reached, should not be sent
+        CPPUNIT_ASSERT_EQUAL(i, mockSentryHandler.sentryUploadedEventCount());
+
+        std::this_thread::sleep_for(std::chrono::seconds(1)); // Wait less than upload rate limit (2s in test)
```

Comment:
Can we lower the rate limit the minimum so as to speed up this test as much as possible?

Reply:
Here is a commit where I set the rate limit to 1s: [0da0908](https://github.com/Infomaniak/desktop-kDrive/pull/291/commits/0da0908d780918bef512ca89763594501931c988)